agreement/models/agreement.py

```diff
@@ -16,8 +16,7 @@ class Agreement(models.Model):
         domain=[('parent_id', '=', False)])
     company_id = fields.Many2one(
         'res.company', string='Company',
-        default=lambda self: self.env['res.company']._company_default_get(
-            'agreement'))
+        default=lambda self: self.env['res.company']._company_default_get())
```

Comment:
Wouldn't `self.env.user.company_id` be cleaner? Also, I see for `git blame`ing several uses of this and the usage of `self.env.user.company_id` that `_company_default_get()` is pretty much the old way of getting the user's company.

Reply:
You may override _company_default_get not self.env.user.company_id

It's a native method and not defined as deprecated